engine/access/rest/converter.go

```diff
@@ -3,21 +3,201 @@ package rest
 import (
 	"encoding/base64"
 	"fmt"
+	"regexp"
 
 	"github.com/onflow/flow-go/engine/access/rest/generated"
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Converter provides the conversion function to/from the Swagger object to Flow objects
+// Converter provides functionality to convert from request models generated using
+// open api spec and flow models.
 
-func toBlock(flowBlock *flow.Block) *generated.Block {
+// Flow section - converting request data to flow models with validation.
+
+func toID(id string) (flow.Identifier, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{64}$`, id)
+	if !valid {
+		return flow.Identifier{}, fmt.Errorf("invalid ID")
```

Comment:
nit: Do we want to add some more details here?

Also, `fmt.Errorf` is intended for use when the error string needs to be a format string. Here it is just a static string, so we can use `errors.New`

Reply:
Also, `HexStringToIdentifier` already returns an error, perhaps we can use that?

```
id, err := flow.HexStringToIdentifier(idStr)
if err != nil {
    return flow.Identifier{}, fmt.Errorf("invalid ID: %w", err)
}
return id, nil
```